src/serialmonitor/serialMonitor.ts

```diff
@@ -165,6 +173,16 @@ export class SerialMonitor implements vscode.Disposable {
         this.serialMonitorApi.dispose();
     }
 
+    private checkForUndefinedSerialMonitorApi(showError: boolean = false): void {
+        const errorString = "Serial Monitor API was not retrieved. You may not have the most recent version of the Serial Monitor extension installed.";
+        if (this.serialMonitorApi === undefined) {
+            Logger.error(errorString)
+            if (showError) {
+                vscode.window.showErrorMessage(errorString);
+            }

```

Comment:
You can use `Logger.notifyUserError()` and `Logger.traceError()` here to get notifications and telemetry for free. Note that `notifyUserError()` will automatically also fire telemetry, so there's no need to call both.

Reply:
Good idea, I'll switch

Reply:
Updated.